### ISSUE TITLE
pylint 1.5.2 ignores decorators on methods

### ISSUE THREAD

**Issue by mikekap** [PyCQA/pylint#744]
Running lint on
```python
def decorate(f):
    f.hello = 'yes'
    return f

class Aclass(object):
    @decorate
    def fn(self):
        print 1

@decorate
def fn():
    print 1

def main():
    a = Aclass()
    print a.fn.hello
    a.fn.goodbye('no')
    print fn.hello
    fn.goodbye('no')

main()
```
Gives
```
E: 16,10: Method 'fn' has no 'hello' member (no-member)
E: 17, 4: Method 'fn' has no 'goodbye' member (no-member)
```

Seems pylint gives up quickly on the function decorator, but for some reason does not for the method.

It would be nice if it actually descended into the decorator, but at least not giving an error would be good. This is a regression from 1.4.5, which does not emit errors for the code above.

**Comment by The-Compiler** [comment]
This might be a duplicate of #659.

**Comment by PCManticore** [comment]
Closed as duplicate.

**Comment by lmazuel** [comment]
@The-Compiler @PCManticore This is not a duplicate anymore, issue #659 as been closed with a fix specific to `lru_cache`, which is not the purpose of this issue here. I found this issue exactly because I got the same problem.

astroid (1.6.1)
pylint (1.8.2)

**Comment by lmazuel** [comment]
Ok, duplicate as this one is fine :)
https://github.com/PyCQA/pylint/issues/1127
